res/plugins/README.md

```diff
@@ -0,0 +1,139 @@
+Tropy Plugins
+=============
+Tropy can be extended via plugins. A plugin is a [CommonJS JavaScript
+module](https://nodejs.org/api/modules.html), which follows the conventions
+outlined in this document to [hook](#hooks) into several aspects of Tropy and
+provide extra functionality.
+
+For a detailed guide on writing your own plugin, please refer to our [sample
+plugin](https://github.com/tropy/tropy-plugin-example).
+
+Installation
+------------
+When installed, Tropy plugins are placed in the plugins folder (alternatively,
+plugins can be installed via `npm` into a `node_modules` sub-folder) accessible
+from Tropy's help menu.
+
+Specification
+-------------
+Like most CommonJS modules, a Tropy plugin consists of a folder with a
+[package.json](https://docs.npmjs.com/files/package.json) file which defines
+the plugin's metadata, configuration, and entry points. In addition to its
+standard properties, this file must include a top-level field "hooks" that
+specifies which Tropy functionality the plugin wants to extend.
+```json
+{
+  "name": "my-plugin",
```

Comment:
worth mentioning productName and description keys in here, with a note that they will show up in the preferences UI?

Reply:
it might be? I considered adding it to the example but not making any mention of it. I would hope for anyone that cares about it they would recognize where the text comes from.

Reply:
I think adding the keys to the example would be useful, but it doesn't need any additional text - like the icon key

Reply:
Equally just including them in the example plugin is fine, whatever you think